docs: Change `--indent-string`'s whitespace representation

rogersheu opened the pull request:
<!--
Thank you for submitting a PR to pylint!

To ease the process of reviewing your PR, do make sure to complete the following boxes.

- [X] Document your change, if it is a non-trivial one.
  - A maintainer might label the issue ``skip-news`` if the change does not need to be in the changelog.
  - Otherwise, create a news fragment with ``towncrier create <IssueNumber>.<type>`` which will be
    included in the changelog. ``<type>`` can be one of the types defined in `./towncrier.toml`.
    If necessary you can write details or offer examples on how the new change is supposed to work.
  - Generating the doc is done with ``tox -e docs``
- [X] Relate your change to an issue in the tracker if such an issue exists (Refs #1234, Closes #1234)
- [X] Write comprehensive commit messages and/or a good description of what the PR does.
- [X] Keep the change small, separate the consensual changes from the opinionated one.
  Don't hesitate to open multiple PRs if the change requires it. If your review is so
  big it requires to actually plan and allocate time to review, it's more likely
  that it's going to go stale.
- [X] If you used multiple emails or multiple names when contributing, add your mails
      and preferred name in ``script/.contributors_aliases.json``
-->

## Type of Changes

<!-- Leave the corresponding lines for the applicable type of change: -->

|     | Type                   |
| --- | ---------------------- |
| ✓   | :scroll: Docs          |

## Description


<!-- If this PR fixes an issue, use the following to automatically close when we merge: -->

It was noticed that when the `all-options.rst` gets viewed on docs, multiple spaces get compressed into one space. This was occurring both in the `help` and in the `default` setting.

To get around this in the `help` section, I used `U+2002`, which is an En Space. I'm flexible on which whitespace character to use, provided it remedies any future confusion.

For the `default` value, I just left it as a regular space, but added quotation marks so the spacing is clear.

Closes #8392

rogersheu committed:
change spaces to U+2002

rogersheu committed:
change spaces to U+2002

rogersheu committed:
add change to changelog

rogersheu committed:
changed the default back to spaces (with "")

rogersheu committed:
changed the default back to spaces (with "")

Pierre-Sassoulas commented on pylint/checkers/format.py:
```diff
@@ -219,11 +219,11 @@ class FormatChecker(BaseTokenChecker, BaseRawFileChecker):
             "indent-string",
             {
-                "default": "    ",
+                "default": '"    "',
```
This change thé default value in the code, it's going to cause problem. There's a mechanism in place to modify the default value for the doc but not the code, see enchent dict of available language for the spell checker or default python interpreter. (On mobile, can't link to the exact place Sorry)

rogersheu replied:
I see. Yeah, I can see how this might cause a problem.

I'm looking at the arguments for enchant. Is this the one you had in mind?

```    options = (
        (
            "spelling-dict",
            {
                "default": "",
                "type": "choice",
                "metavar": "<dict name>",
                "choices": _get_enchant_dict_choices(enchant_dicts),
                "help": _get_enchant_dict_help(enchant_dicts, PYENCHANT_AVAILABLE),
            },
        ),
...
```

I see what's happening here, with functions being passed into choices/help, but the real issue is that `default` has to be four spaces, `   `, exactly in Python. This means that the value has to be different when interpreted for the conversion to `reStructuredText`.

Does adding such functionality sound like a reasonable path forward for this PR?

Pierre-Sassoulas replied:
Back on a PC 😄 I meant this:

https://github.com/pylint-dev/pylint/blob/eb33f8a6c0d51be08a6e3ae6a5bff0350c4048a5/doc/exts/pylint_options.py#L41-L48

rogersheu replied:
Thanks for pointing me to this @Pierre-Sassoulas !!

Been busy with work lately, but I'm hoping to find a little time to test out this change soon. 😅

Pierre-Sassoulas reviewed (changes requested):
Thank you the pull request, there's no need for a changelog here I applied the skip News label :)

Pierre-Sassoulas committed:
Move the indent-string doc fix to the doc-only override mechanism

Rendering the --indent-string default and help in all-options.rst
collapses the consecutive spaces, hiding the actual 4-space value.

Instead of changing the real option default and the CLI --help text
(which would alter runtime behavior and put ambiguous-unicode spaces in
the help output), express the fix purely in the docs generator via
DYNAMICALLY_DEFINED_OPTIONS: the default is shown quoted and the help
uses en spaces (U+2002) so the spacing stays visible. The TOML example
unquotes the dynamic default so it still shows the real value.

Closes #8392

Pierre-Sassoulas committed:
Escape the indent-string tab so it renders as \t not bare t

reStructuredText treats a lone backslash as an escape character, so the
"\t" in the --indent-string help was rendered as a bare "t" on the
options page. Double the backslash in the doc-only override so it shows
as "\t" again.
